facia/app/controllers/FaciaController.scala

```diff
@@ -240,41 +241,47 @@ trait FaciaController
           )(request),
           targetedTerritories,
         )
+      case Some((faciaPage: PressedPage, targetedTerritories)) if request.isRss =>
```

Comment:
I have been thinking `renderFrontPressResult`  method could do with some refactoring for some time now as it is quite long and does too many things. Feel free to skip this suggestion for now though as it is not directly related to your change but had an idea as I was reading your PR.

We could have two methods:

```
private def resultWithVaryHeader(result: CacheableResult, targetedTerritories: Boolean)(implicit request: RequestHeader)= {
      withVaryHeader(successful(Cached(CacheTime.Facia)(result)), 
      targetedTerritories)
  }
  
 private def resultWithVaryAndPreloadHeader(faciaPage: PressedPage, targetedTerritories: Boolean)(implicit request: RequestHeader) = {
    withVaryHeader(
      successful(
        Cached(CacheTime.Facia)(RevalidatableResult.Ok(FrontHtmlPage.html(faciaPage)))
          .withPreload(
            Preload.config(request).getOrElse(context.applicationIdentity, Seq.empty),
          )(context, request)
          .withPreconnect(HttpPreconnections.defaultUrls),
      ),
      targetedTerritories,
    )
  }

    
```

and call them inside `renderFrontPressResult` as needed, e.g.:

```
        } else JsonFront(faciaPage)
        resultWithVaryHeader(result, targetedTerritories)
      case Some((faciaPage: PressedPage, targetedTerritories)) if request.isEmail || ConfigAgent.isEmailFront(path) =>
        resultWithVaryHeader(renderEmail(faciaPage), targetedTerritories)
      case Some((faciaPage: PressedPage, targetedTerritories)) if TrailsToShowcase.isShowcaseFront(faciaPage) =>
        resultWithVaryHeader(renderShowcaseFront(faciaPage), targetedTerritories)
      case Some((faciaPage: PressedPage, targetedTerritories)) =>
        resultWithVaryAndPreloadHeader(faciaPage, targetedTerritories)
```

`renderFrontPressResult` would still be long but that would be a start! Like I said feel free to merge without this!

Reply:
I think that's a great suggestion! I'll try it out and see how it feels!